### DISCUSSION TITLE
Can I use Nodemailer on Nuxt 3?

### DISCUSSION THREAD

**Discussion by phyesix** [nuxt/nuxt#16834] [Questions] [answered]
Hi,

Thank you for this amazing community.

When I looked at the documentation for **Nuxt 3,** I saw that we can write an API with a folder named **server**. Is it possible to create an API for a simple contact page and send mail using nodemail or another alternative ways? I couldn't find any example.

**Comment by TheAlexLichter** [comment, marked as answer]
Yes, this is absolutely possible. Thanks to `h3` you have the full power of a server in Nitro and thus Nuxt3.

Untested example:


```js
// server/api/contact.post.js

import nodemailer from 'nodemailer'

// Or use SMTP if you feel like
const transporter = nodemailer.createTransport({
  sendmail: true,
  newline: 'unix',
  path: '/usr/sbin/sendmail'
})

export default defineEventHandler(async (event) => {
  const body = await readBody(event) // Pass params via JSON in the POST request

  transporter.sendMail({
    from: body.email,
    to: 'CHANGE_THIS@mail.com',
    subject: 'New contact form message',
    text: body.text
  })
})
```

**Reply by thaikolja** [reply]
Here is another example as a tiny, light project with Nodemailer function and lots of comments. Maybe it helps someone in the future:

https://gitlab.com/thaikolja/nuxt-nodemailer-example
